### ISSUE TITLE
Missing classes when processing Groovydoc

### ISSUE THREAD

**Issue by nre-ableton** [Kapeli/javadocset#10]
(Migrated from https://github.com/Kapeli/javadocset/issues/9#issuecomment-473287072)

When processing Groovydoc-generated documentation, some classes are missing in the listing. Here's a screenshot of an [example project](https://github.com/AbletonDevTools/python-pipeline-utils) which demonstrates the behavior:

<img width="256" alt="Screen Shot 2019-03-15 at 12 45 37" src="https://user-images.githubusercontent.com/16208985/54434673-100f8b80-472f-11e9-9bab-cb3b9f438c56.png">

Here's the corresponding list of classes/singletons from the library:

```
$ tree -P "*.groovy" src/ vars/
src/
└── com
    └── ableton
        ├── Pipenv.groovy
        ├── PythonBuilder.groovy
        ├── PythonPackage.groovy
        ├── VirtualEnv.groovy
        └── VirtualEnvList.groovy
vars/
├── pipenv.groovy
├── pythonBuilder.groovy
├── pythonPackage.groovy
├── virtualenv.groovy
└── virtualenvs.groovy
```

Looking at https://github.com/Kapeli/javadocset/blob/master/javadocset/DHIndexer.m#L63, `javadocset` is looking for the words `Class in` within `index-all.html`, and `groovydoc` seems to be at fault here.

I did a bit of investigating, and it seems that `groovydoc` is parsing classes before scripts, and if it encounters a script with the same (case-insensitive) name as a script, the script takes precedence (or perhaps overwrites the class entry). That's why in the above example, only `VirtualEnvList` is shown in classes, since it is the only class that lacks a singleton. In all of the other cases, the class has a corresponding singleton which differs only by case (ie, the 'pipenv` script vs. the `Pipenv` class).

A possible solution here would be to derive the class/script entries by parsing the `allclasses-frame.html` file instead.

**Comment by Kapeli** [comment]
This is kind of an obscure issue that I haven't encountered before, so I'd rather not add something special for it. If more users report this issue, I'll try to work around it.

`allclasses-frame.html` isn't suitable, as it also lists things that aren't classes (i.e. the scripts).

My recommendation would be to figure out a way to get classes to show up in `index-all.html`.

**Comment by nre-ableton** [comment]
Fair enough -- I'll see if there's either a way to make `groovydoc` behave more nicely, or try to report a bug there. Feel free to "wontfix" this issue if you like.
